### ISSUE TITLE
Test System.Net.Sockets.Tests.*.SendRecvPollSync_TcpListener_Socket(listenAt: ::1, pollBeforeOperation: True) failed in CI.

### ISSUE THREAD

**Issue by Sunny-pu** [dotnet/runtime#1479]
**Edit** by @antonfirsov: recent failures seem to happen on Windows only, see https://github.com/dotnet/runtime/issues/1479#issuecomment-750698738.

---
Test: System.Net.Sockets.Tests.SendReceiveMemoryNativeTask.SendRecvPollSync_TcpListener_Socket(listenAt: ::1, pollBeforeOperation: True) has failed.

MESSAGE:
Assert.Equal() Failure\nExpected: 123456\nActual: 123455
+++++++++++++++++++
STACK TRACE:
at System.Net.Sockets.Tests.SendReceive`1.SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, Boolean pollBeforeOperation) in /mnt/j/workspace/dotnet_corefx/master/outerloop_netcoreapp_ubuntu16.04_debug/src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs:line 630 --- End of stack trace from previous location where exception was thrown ---

Details:
https://ci.dot.net/job/dotnet_corefx/job/master/job/outerloop_netcoreapp_ubuntu16.04_debug/403/testReport/System.Net.Sockets.Tests/SendReceiveMemoryNativeTask/SendRecvPollSync_TcpListener_Socket_listenAt____1__pollBeforeOperation__True_/

**Comment by v-haren** [comment]
failed again in job: [runtime-libraries-coreclr outerloop 20200909.3 ](https://dev.azure.com/dnceng/public/_build/results?buildId=806988&view=ms.vss-test-web.build-test-results-tab&runId=25591310&paneView=debug&resultId=103032)

failed test: System.Net.Sockets.Tests.SendReceiveApm.SendRecvPollSync_TcpListener_Socket(listenAt: ::1, pollBeforeOperation: False)

net5.0-Windows_NT-Release-x64-CoreCLR_release-Windows.81.Amd64.Open


Error message
~~~
Read poll after completion should have succeeded
Expected: True
Actual:   False


Stack trace
   at System.Net.Sockets.Tests.SendReceive`1.<>c__DisplayClass10_0.<<SendRecvPollSync_TcpListener_Socket>b__0>d.MoveNext() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs:line 605
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 96
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 124
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 84
   at System.Net.Sockets.Tests.SendReceive`1.SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, Boolean pollBeforeOperation) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs:line 645
--- End of stack trace from previous location ---
~~~

**Comment by v-haren** [comment]
failed again in job: [runtime-libraries-coreclr outerloop 20201223.2 ](https://dev.azure.com/dnceng/public/_build/results?buildId=930428&view=ms.vss-test-web.build-test-results-tab&runId=29477834&resultId=103100&paneView=debug)

failed test: System.Net.Sockets.Tests.SendReceive_Sync.SendRecvPollSync_TcpListener_Socket(listenAt: 127.0.0.1, pollBeforeOperation: False)

net6.0-windows-Release-x64-CoreCLR_release-(Windows.Nano.1809.Amd64.Open)windows.10.amd64.serverrs5.open@mcr.microsoft.com/dotnet-buildtools/prereqs:nanoserver-1809-helix-amd64-08e8e40-20200107182504

Error message
~~~
Read poll after completion should have succeeded
Expected: True
Actual:   False


Stack trace
   at System.Net.Sockets.Tests.SendReceive`1.<>c__DisplayClass7_0.<<SendRecvPollSync_TcpListener_Socket>b__0>d.MoveNext() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 521
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 96
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 124
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 90
   at System.Net.Sockets.Tests.SendReceive`1.SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, Boolean pollBeforeOperation) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 575
--- End of stack trace from previous location ---
~~~

**Comment by antonfirsov** [comment]
Removed `os-linux` label, since all recent failures seem to happen on Windows according to Kusto.

**Comment by antonfirsov** [comment]
Recent failures are happening in sync variants.

**Comment by VincentBu** [comment]
Similar test failed in [runtime-libraries-coreclr outerloop 20210401.4](https://dev.azure.com/dnceng/public/_build/results?buildId=1067573&view=ms.vss-test-web.build-test-results-tab&runId=32839632&resultId=103537&paneView=debug)

Failed test:
```
net6.0-windows-Release-x64-CoreCLR_release-Windows.81.Amd64.Open
 -System.Net.Sockets.Tests.SendReceive_SpanSyncForceNonBlocking.SendRecvPollSync_TcpListener_Socket(listenAt: ::1, pollBeforeOperation: False)
```

**Error message:**
```
Read poll after completion should have succeeded
Expected: True
Actual:   False


Stack trace
   at System.Net.Sockets.Tests.SendReceive`1.<>c__DisplayClass7_0.<<SendRecvPollSync_TcpListener_Socket>b__0>d.MoveNext() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 521
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 63
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 82
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 49
   at System.Net.Sockets.Tests.SendReceive`1.SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, Boolean pollBeforeOperation) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 575
--- End of stack trace from previous location ---
```

**Comment by VincentBu** [comment]
Failed again in [runtime-libraries-coreclr outerloop 20210411.4](https://dev.azure.com/dnceng/public/_build/results?buildId=1082333&view=ms.vss-test-web.build-test-results-tab&runId=33190922&resultId=103608&paneView=debug)

Faield test:
```
net6.0-windows-Release-x64-CoreCLR_release-Windows.81.Amd64.Open
 -System.Net.Sockets.Tests.SendReceive_SpanSync.SendRecvPollSync_TcpListener_Socket(listenAt: ::1, pollBeforeOperation: False)
```

**Error message:**
```
Read poll after completion should have succeeded
Expected: True
Actual:   False


Stack trace
   at System.Net.Sockets.Tests.SendReceive`1.<>c__DisplayClass7_0.<<SendRecvPollSync_TcpListener_Socket>b__0>d.MoveNext() in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 521
--- End of stack trace from previous location ---
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 63
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 82
   at System.Threading.Tasks.TaskTimeoutExtensions.WhenAllOrAnyFailed(Task[] tasks, Int32 millisecondsTimeout) in /_/src/libraries/Common/tests/System/Threading/Tasks/TaskTimeoutExtensions.cs:line 49
   at System.Net.Sockets.Tests.SendReceive`1.SendRecvPollSync_TcpListener_Socket(IPAddress listenAt, Boolean pollBeforeOperation) in /_/src/libraries/System.Net.Sockets/tests/FunctionalTests/SendReceive/SendReceive.cs:line 575
--- End of stack trace from previous location ---
```

**Comment by MihaZupan** [comment]
Disabled for Windows 8 in #51186

**Comment by wfurt** [comment]
triage: we should re-enable the test and fix it if we see failures.

**Comment by karelz** [comment]
Fixed in #79145
